crates/ruff_linter/src/rules/pyflakes/rules/undefined_name.rs

```diff
@@ -19,17 +19,36 @@ use ruff_macros::{derive_message_formats, violation};
 ///     return n * 2
 /// ```
 ///
+/// ## Options
+/// - [`target-version`]: Can be used to configure which symbols Ruff will understand
+///   as being available in the `builtins` namespace.
+///
 /// ## References
 /// - [Python documentation: Naming and binding](https://docs.python.org/3/reference/executionmodel.html#naming-and-binding)
 #[violation]
 pub struct UndefinedName {
     pub(crate) name: String,
+    pub(crate) minor_version_builtin_added: Option<u8>,
 }
 
 impl Violation for UndefinedName {
     #[derive_message_formats]
     fn message(&self) -> String {
-        let UndefinedName { name } = self;
-        format!("Undefined name `{name}`")
+        let UndefinedName {
+            name,
+            minor_version_builtin_added,
+        } = self;
+        let tip = minor_version_builtin_added.map(|version_added| {
+            format!(
+                "Added as a builtin in Python 3.{version_added}; \
+consider specifying a newer `target-version` in your Ruff config."
+            )
+        });
+
+        if let Some(tip) = tip {
+            format!("Undefined name `{name}`. {tip}")
+        } else {
+            format!("Undefined name `{name}`")
+        }
     }
```

Comment:
This could be done as a single `if-let`, but we get nicer docs if we split it up like this (the generated summary message in the table is _very_ long otherwise):

![image](https://github.com/user-attachments/assets/aa7ebda1-8b8c-4c7e-8c6d-bfa0e99b2cda)